go/libkb/gpg_key.go

```diff
@@ -39,8 +42,19 @@ func (g *GPGKey) GetAlgoType() AlgoType {
 }
 
 func (g *GPGKey) SignToString(msg []byte) (sig string, id keybase1.SigID, err error) {
-	g.G().Log.Debug("GPGKey Signing %s", string(msg))
-	sig, err = g.G().GetGpgClient().Sign(*g.fp, msg)
+	g.G().Log.Debug("+ GPGKey Signing %s", string(msg))
+	defer func() {
+		g.G().Log.Debug("- GPGKey Signing -> %s", err)
+	}()
+
+	if g.ct == keybase1.ClientType_CLI {
+		g.G().Log.Debug("| GPGKey reverse delegate to CLI")
+		sig, err = g.ui.Sign(context.TODO(), keybase1.SignArg{Fingerprint: (*g.fp)[:], Msg: msg})
+	} else {
+		g.G().Log.Debug("| GPGKey sign in-process; let's hope for the best!")
```

Comment:
For Electron, we need to fork the `gpg` CLI ourselves, since reflecting back to the client won't help.